preprocessy/outliers/_handleoutlier.py

```diff
@@ -37,23 +37,13 @@ def __init__(self, train_df, params):
             -q3 : specify the end of the range (Default is 0.95)
 
         """
-        self.train_df = train_df
+        self.train_df = None
         self.cols = []
         self.remove_outliers = True
         self.replace = False
         self.quartiles = {}
         self.first_quartile = 0.05
         self.third_quartile = 0.95
-        if "cols" in params.keys():
-            self.cols = params["cols"]
-        if "removeoutliers" in params.keys():
-            self.remove_outliers = params["removeoutliers"]
-        if "replace" in params.keys():
-            self.replace = params["replace"]
-        if "q1" in params.keys():
-            self.first_quartile = params["q1"]
-        if "q3" in params.keys():
-            self.third_quartile = params["q3"]
```

Comment:
You have to move this to the beginning of `handle_outliers` otherwise nothing will be initialised.

Reply:
Okay, will do.